### ISSUE TITLE
Removal of short sequences produces more ASVs, longer final sequences and retains more reads. Why?

### ISSUE THREAD

**Issue by gcuster1991** [benjjneb/dada2#1157]
Hi @benjjneb and others,
First off, thank you for maintaining such an active help community here. I apologize if this has been addressed in another issue that I did not see. I'm trying to understand why removal of short sequences during the filterandTrim step (minLen=X) produces more ASVs, retains a larger percentage of reads through the pipeline and produces more long reads in the final sequence table. As some background, we have sequenced the ITS1 region (208+/-50bp) on a NovaSeq with 2x300 reads. It appear as if our results have many short sequences and inclusion of these reads seems to greatly reduce the number of ASV calls as well as the number of sequences retained through the processing pipeline. 

We are using dada v- 1.16.0, R3.5, and following the tutorial found at (https://benjjneb.github.io/dada2/tutorial_1_8.html).

I've created a summary table for several different cutoffs for minimum read lengths created using the below command. All other parts of the processing pipeline remained constant. 

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncQ=19, minLen=100,
              maxN=0, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE, verbose=TRUE)

Column header info. Initial trim method (min length of read to be included) | #ASVs | % total reads retained | Mean and median read length of final representative seqs | % retained reads assigned to class level

**Min read length       #ASV,    %retained,            Mean&median,    %assigned to class**
No min length       | 750      | 8.4%                    | 143 & 148          |  < 1%
Min length of 50   | 1739    | 29%                     | 222 & 203         | 42.5%
Min length of 75   | 1885    | 46.51%                | 256 & 263         | ~47%
Min length of 100 | 1862    | 40.6%                  | 269 & 276         | ~47%
Min length 125     | 1773     | 34.2%                  | 284 & 280         | 59.5%
Min length of 150 | 3829   | 27.14                    | 304 & 312          | 27.37%

Any insight would be greatly appreciated.

**Comment by benjjneb** [comment]
For ITS data, especially on 2x300 sequences for the ITS1 region -- which varies a lot in length and is <300 bps itself -- I highly recommend using the ITS-specific tutorial. This leverages cutadapt to identify and remove primers prior to processing by DADA2, and should improve your results over any of the options you are investigating here.

You could also consider using just your forward reads given that they will completely cover the ITS1 region, see here for benchmarking supporting that choice: https://doi.org/10.1016/j.funeco.2019.03.005

**Comment by gcuster1991** [comment]
@benjjneb, thank you for the quick response. Would you suggest removing primers even if degenerate primers are used to capture additional diversity in the flanking regions? Wouldn't discard information that may be useful?

**Comment by benjjneb** [comment]
> Would you suggest removing primers even if degenerate primers are used to capture additional diversity in the flanking regions?

Yes, it's an absolute must to remove degenerate primers.

Remember, the primer sequence is not the biological sequence, and more than one primer sequence can and will bind sufficiently to the same biological sequence to produce amplicons. The primer sequence on the reads is technical diversity introduce by the library preparation, not real diversity.

**Comment by gcuster1991** [comment]
@benjjneb, Thank you. I had a suspicion this might have been contributing to my issues. However, I was being optimistic that the increased diversity in that region might be useful. I'll follow up once I have removed primers and reran the processing.

**Comment by gcuster1991** [comment]
Hi @benjjneb, I've had a chance to remove primers and rerun this processing for my dataset with two different minLen parameters (0 and 75). The results still produce are similar. When short reads are excluded from the dataset, more ASVs are assigned. I'm trying to understand why this is the case. Any thoughts?

**Comment by benjjneb** [comment]
> When short reads are excluded from the dataset, more ASVs are assigned. I'm trying to understand why this is the case. Any thoughts? Typically that is not expected.

My first thought... low length (and quality) sequences are messing up the error model learneing step. What happens if you remove that difference, by using the same error model for the denoising of each read set?

**Comment by gcuster1991** [comment]
Sorry, I'm not sure I understand what you mean by "remove that difference, by using the same error model for the denoising of each read set". Could you clarify?

**Comment by benjjneb** [comment]
Let's say `filtFs.XX` corresponds to your filtered forward read files with `minLen=XX` enforced. You can then learn an error model from the full set of filtered reads (`filtFs.00`) and the filtered reads with short reads removed (`filtFs.75`).

First I'd just plot the error models learned from each:
```
err.00 <- learnErrors(filtFs.00, multi=TRUE)
err.75 <- learnErrors(filtFs.75, multi=TRUE)
plotErrors(err.00, nominalQ=TRUE)
plotErrors(err.75, nominalQ=TRUE)
```

Do they look different?

Then I would try denoising the same set of filtered files with the two error models:
```
dd.00.err00 <- dada(filtFs.00, err=err.00)
dd.00.err75 <- dada(filtFs.00, err=err.75)
```

Are there quire a few more ASVs being detected from the same files (`filtFs.00`) by one error model versus the other? That would indicate the issue lies in how the short reads are affecting the learning of the error model.

**Comment by gcuster1991** [comment]
I ran the suggested analysis with ~20 samples to speed up the process, so the number of ASVs is smaller than reported above.

The error plots look similar. 
Error with no minlen:
![Forward](https://user-images.githubusercontent.com/24256789/96517475-3a3c0000-1226-11eb-87f3-5e5957a13a10.jpeg)
Error with 75 bp minlen:
![Forward](https://user-images.githubusercontent.com/24256789/96517534-55a70b00-1226-11eb-83a8-435da8c5438f.jpeg)

As for your second suggestion:

dd.00.err00 <- dada(filtFs.00, err=err.00): Produced 611 ASVs.

dd.00.err75 <- dada(filtFs.00, err=err.75): Produced 815 ASVs.

So, my interpretation is that since the number of ASVs assigned is ~30% different, the short reads are influencing the error modeling. Moving forward, I think short reads should be removed prior to processing? Would you agree?

**Comment by gcuster1991** [comment]
I ran the suggested analysis with ~20 samples to speed up the process, so the number of ASVs is smaller than reported above.
The error plots look similar. Error with no minlen: 
￼
![Forward](https://user-images.githubusercontent.com/24256789/96521212-01a02480-122e-11eb-9403-7c6de134a565.jpeg)

Error with 75 bp minlen: 
![Forward](https://user-images.githubusercontent.com/24256789/96521236-0f55aa00-122e-11eb-81d8-d5aab5b1578b.jpeg)
￼

As for your second suggestion:
dd.00.err00 <- dada(filtFs.00, err=err.00): Produced 611 ASVs.
dd.00.err75 <- dada(filtFs.00, err=err.75): Produced 815 ASVs.
So, my interpretation is that since the number of ASVs assigned is ~30% different, the short reads are influencing the error modeling. Moving forward, I think short reads should be removed prior to processing? Would you agree?

**Comment by benjjneb** [comment]
> So, my interpretation is that since the number of ASVs assigned is ~30% different, the short reads are influencing the error modeling. Moving forward, I think short reads should be removed prior to processing? Would you agree?

Yes, I agree with your interpretation.

**Comment by gcuster1991** [comment]
Thanks @benjjneb . I'll close this for now.

**Comment by gcuster1991** [comment]
Upon further digging, I realized this problem is due to the trunQ parameter and binnning of NovaSeq quality scores. With NovaSeq's binned quality, even though the mean appears to be very high, there were single basepairs throughout out the reads that dipped below the quality of 22 (? second highest bin, I used trunQ=19 to be below this threshold). This led to many truncated short reads and a density of reads related to read length (many short and few long). I addressed this by making the trunQ parameter less strict (trunQ=10) and implementing a maxEE=c(2,2) filtering instead. This allowed me to retain a larger number of reads as well as keep the mean and median read length closer to what was expected.
